code/datums/components/crafting/recipes.dm

```diff
@@ -1150,3 +1153,120 @@
 				/obj/item/aquarium_kit = 1
 				)
 	category = CAT_MISC
+
+/datum/crafting_recipe/alcohol_burner
+	name = "Alcohol burner"
+	result = /obj/item/burner
+	time = 5 SECONDS
+	reqs = list(/obj/item/reagent_containers/glass/beaker  = 1,
+				/datum/reagent/consumable/ethanol = 15,
+				/obj/item/paper = 1
+				)
+	category = CAT_CHEMISTRY
+
+/datum/crafting_recipe/oil_burner
+	name = "Oil burner"
+	result = /obj/item/burner/oil
+	time = 5 SECONDS
+	reqs = list(/obj/item/reagent_containers/glass/beaker  = 1,
+				/datum/reagent/fuel/oil = 15,
+				/obj/item/paper = 1
+				)
+	category = CAT_CHEMISTRY
+
+/datum/crafting_recipe/fuel_burner
+	name = "Fuel burner"
+	result = /obj/item/burner/fuel
+	time = 5 SECONDS
+	reqs = list(/obj/item/reagent_containers/glass/beaker  = 1,
+				/datum/reagent/fuel = 15,
+				/obj/item/paper = 1
+				)
+	category = CAT_CHEMISTRY
+
+/datum/crafting_recipe/thermometer
+	name = "Thermometer"
+	tools = list(TOOL_WELDER)
+	result = /obj/item/thermometer
+	time = 5 SECONDS
+	reqs = list(
+				/datum/reagent/mercury = 5,
+				/obj/item/stack/sheet/glass = 1
+				)
+	category = CAT_CHEMISTRY
+
+/datum/crafting_recipe/thermometer_alt
+	name = "Thermometer"
+	result = /obj/item/thermometer/pen
+	time = 5 SECONDS
+	reqs = list(
+				/datum/reagent/mercury = 5,
+				/obj/item/pen = 1
+				)
+	category = CAT_CHEMISTRY
+
+/datum/crafting_recipe/ph_booklet
+	name = "pH booklet"
+	result = /obj/item/ph_booklet
+	time = 5 SECONDS
+	reqs = list(
+				/datum/reagent/universal_indicator = 5,
+				/obj/item/paper = 1
+				)
+	category = CAT_CHEMISTRY
+
+/datum/crafting_recipe/dropper //Maybe make a glass pipette icon?
+	name = "Dropper"
+	result = /obj/item/reagent_containers/dropper
+	tools = list(TOOL_WELDER)
+	time = 5 SECONDS
+	reqs = list(
+				/obj/item/stack/sheet/glass  = 1,
+				)
+	category = CAT_CHEMISTRY
+
+/datum/crafting_recipe/improvised_chem_heater
+	name = "Improvised chem heater"
+	result = /obj/machinery/space_heater/improvised_chem_heater
+	tools = list(TOOL_SCREWDRIVER, TOOL_MULTITOOL, TOOL_WIRECUTTER)
+	time = 15 SECONDS
+	additional_req_text = " 1 adjacent space heater"
+	reqs = list(
+				/obj/item/stack/cable_coil = 2,
+				/obj/item/stack/sheet/glass = 2,
+				/obj/item/stack/sheet/iron = 2,
+				/datum/reagent/water = 50,
+				/obj/item/thermometer = 1
+				)
+	category = CAT_CHEMISTRY
+	var/obj/cached_heater
+
+/datum/crafting_recipe/improvised_chem_heater/check_requirements(mob/user, list/collected_requirements)
+	. = ..()
+	var/obj/machinery/space_heater/s_heater = locate(/obj/machinery/space_heater) in orange(user, 1)
+	if(!s_heater)
+		return FALSE
+	cached_heater = s_heater
+	return TRUE
```

Comment:
This will cause a hard delete due to improperly handled reference.
In any case, this needs to be handled more generically.
Modify `get_surroundings()` to scan for objects in general, not just items, instead.

Reply:
Okay, so I modified recipes to allow use of machines - only issues is, the battery is still spawned after the qdel of the machine, and thus you get a free battery. How would you recommend addressing this issue considering my previous (left in as commented) method doesn't quite work here?

Reply:
Without looking at the specific code I assume that the machine is spawning containing a battery without the recipe requiring one.
The sound fix would be to make it spawn without one. A less sound one would be for it to require a battery at creation.
Makes sense?

Reply:
Oh! That's one way to do it

Reply:
This has been redone and fixed